### ISSUE TITLE
URLs are pseudolocalized - should be ignored?

### ISSUE THREAD

**Issue by aormsby** [martincostello/Pseudolocalizer#74]
I've seen some other tools that skip over URLs when generating pseudolocalized text. This makes sense to me since you wouldn't want those to get localized (at least not as a translated string). It might be nice to identify and ignore URLs during the process.

(But hey, I really like your tool. Very simple. Works on my Mac - thrilled about this.)

**Comment by martincostello** [comment]
Hey Adam - thanks for raising this issue, it's definitely not a use case I'd considered before. I'd not come across this myself as in projects I work on URLs are usually treated as format string parameters, rather than being part of the localised string itself.

I'll look into this when I get some time to do so, I already made a similar change to handle HTML tags correctly a while back for similar reasons.

**Comment by aormsby** [comment]
Yeah, I initially thought the use case was a bit odd as well, and I admit I'm not sure what the best practice would be. I thought about it when I was running some tests in a Xamarin project. The starter project I modified used XAML to display a label with a URL as the text value, and I just treated it as a localized string without really thinking about it. In one sense, that could be a good setup since I could set a different URL based on locale if I wanted to, but it also seemed like the situation might not occur that often. In any case, I saw the pseudolocalized URL and thought, "Oh, I don't want to translate that." So that's kind of where my head was. Food for thought, I guess. 🙂

**Comment by martincostello** [comment]
Hi Adam, I've looked into this and it doesn't look particularly trivial to achieve.

Because the transformations are looked at one character at a time with look-ahead, looking for `://` as the URI marker and then backtracking to "undo" things would need me to rework the existing escaping mechanism quite a lot.

Also because of URI schemes (more than just HTTP and HTTPS), reserved characters, escaping/encoding etc. (I got as far as looking into [RFC-3986](https://tools.ietf.org/html/rfc3986#section-2)), there's probably a lot of edge cases that would either be missed and make the solution incomplete, or require a lot of effort to get right.

At least URLs already embedded within HTML tags are already ignored and left alone.

For now I'd just recommend that URLs are stored separately (as really they're data not text, even if they vary by language) and are inserted into the relevant text strings via formatting.
